alns/accept/GreatDeluge.py

```diff
@@ -0,0 +1,63 @@
+from typing import Optional
+
+from alns.accept.AcceptanceCriterion import AcceptanceCriterion
+
+
+class GreatDeluge(AcceptanceCriterion):
+    """
+    The Great Deluge (GD) criterion accepts solutions if the candidate solution
+    has a value lower than a threshold (originally called the water level [1]).
+    The initial threshold is computed as
+
+    ``threshold = alpha * initial.objective()``
+
+    where ``initial`` is the initial solution passed-in to ALNS, inferred
+    from the best solution at the first iteration. The threshold is updated in
```

Comment:
Question: what does "inferred from the best solution at the first iteration" mean _precisely_? Is it not just the initial passed-in solution?

Reply:
Yes, it is the passed-in initial solution. I initially thought it would have mattered, because ‘best’ and ‘current’ both have a reference to the initial solution. Let me make another edit to make this more clear